src/config.ts

```diff
@@ -1,19 +1,37 @@
 import { Logger } from '@opentelemetry/api';
+import { DEFAULT_BREEZE_ENDPOINT } from './Declarations/Constants';
 
-export interface ExporterConfig {
+interface InstrumentationKeyRequired {
+  instrumentationKey: string;
+  connectionString?: undefined;
+}
+
+interface ConnectionStringRequired {
+  instrumentationKey?: undefined;
+  connectionString: string;
+}
+
+export interface AzureExporterConfig {
   // Exporter
-  instrumentationKey?: string;
-  connectionString?: string;
+  logger?: Logger;
 
   // Channel
   batchSendRetryIntervalMs: number;
-  logger?: Logger;
 
   // Sender
   maxConsecutiveFailuresBeforeWarning: number;
+  endpointUrl: string;
 }
 
-export const DEFAULT_EXPORTER_CONFIG: ExporterConfig = {
+export type AzureExporterConfigWithSetupString = AzureExporterConfig &
```

Comment:
Having some comments describing what are this types for would be useful

Reply:
Agreed. I initially wanted to do some fancy `OneOf` assertions. Env vars are a common scenario so it'd be more like `AtModeOne`, which is trickier to add. Better to just simplify this entire file to be a simple config interface